src/pages/RFOX/components/RewardsAndClaims/Rewards.tsx

```diff
@@ -24,93 +26,90 @@ type RewardsProps = {
-        return maybeRuneAddress || mockRfoxRewardsRuneAddress
-      })
-      .filter(isSome)
-      .map(runeAddress => {
-        return toAccountId({
-          chainId: thorchainChainId,
-          account: runeAddress as string,
-        })
-      })
-  }, [runeAddresses])
-
-  const txIdsFilter = useMemo(() => {
-    return {
-      assetId: thorchainAssetId,
-      accountIds: thorchainAccountIds,
-      from: DAO_TREASURY_THORCHAIN,
-    }
-  }, [thorchainAccountIds])
-
-  // TODO: Fetch any rune accounts not held in tx history
-  const txIds = useAppSelector(state => selectReceivedTxsForAccountIdsByFilter(state, txIdsFilter))
+    data: maybeRewardDistributions,
+    isLoading: isRewardDistributionsLoading,
+    isFetching: isRewardDistributionsFetching,
+  } = useLifetimeRewardDistributionsQuery({ stakingAssetAccountAddresses })
 
   const isLoading = useMemo(() => {
-    // TODO: show loading state if tx history is also loading
-    return (
-      isStakingInfoHistoryLoading ||
-      isCurrentStakingInfoLoading ||
-      isStakingInfoHistoryFetching ||
-      isCurrentStakingInfoFetching
+    return isRewardDistributionsLoading || isRewardDistributionsFetching
+  }, [isRewardDistributionsFetching, isRewardDistributionsLoading])
+
+  const rewardDistributionsByTxId = useMemo(() => {
+    if (!maybeRewardDistributions) return {}
+    return maybeRewardDistributions.reduce<Record<string, RewardDistribution>>(
+      (acc, rewardDistribution) => {
+        acc[rewardDistribution.txId] = rewardDistribution
+        return acc
+      },
+      {},
     )
-  }, [
-    isCurrentStakingInfoFetching,
-    isCurrentStakingInfoLoading,
-    isStakingInfoHistoryFetching,
-    isStakingInfoHistoryLoading,
-  ])
+  }, [maybeRewardDistributions])
+
+  const txIds = useMemo(() => {
+    return Object.keys(rewardDistributionsByTxId)
+  }, [rewardDistributionsByTxId])
+
+  const getTxDetails = useCallback(
+    (txId: TxId): TxDetails | undefined => {
+      if (!rune) return
+
+      const { rewardAddress, amount } = rewardDistributionsByTxId[txId]
+
+      const txTransfer: TxTransfer = {
+        from: [DAO_TREASURY_THORCHAIN],
+        to: [rewardAddress],
+        value: amount,
+        assetId: thorchainAssetId,
+        type: TransferType.Receive,
+      }
+
+      const tx: Tx = {
+        pubkey: rewardAddress,
+        status: TxStatus.Confirmed,
+        chainId: thorchainChainId,
+        blockHeight: 0,
+        blockTime: 0,
+        confirmations: 0,
+        txid: txId,
+        transfers: [txTransfer],
+      }
+
+      const txLink = getTxLink({
+        name: Dex.Thor,
+        defaultExplorerBaseUrl: '',
+        txId,
+      })
+
+      return {
+        tx,
+        fee: undefined,
+        transfers: [
+          {
+            ...txTransfer,
+            asset: rune,
+          },
+        ],
+        type: TransferType.Receive,
+        txLink,
+      }
+    },
+    [rewardDistributionsByTxId, rune],
+  )
```

Comment:
This totally negates the dependence of rFOX tx history on the tx history slice, allowing us to display txs for RUNE accounts not connected to the app